lapce-data/src/keypress.rs

```diff
@@ -864,26 +1176,18 @@ impl KeyPressData {
             if parts.is_empty() {
                 continue;
             }
-            let key = match parts[parts.len() - 1].to_lowercase().as_str() {
-                "escape" => druid::KbKey::Escape,
-                "esc" => druid::KbKey::Escape,
-                "space" => druid::KbKey::Character(" ".to_string()),
-                "backspace" => druid::KbKey::Backspace,
-                "bs" => druid::KbKey::Backspace,
-                "arrowup" => druid::KbKey::ArrowUp,
-                "arrowdown" => druid::KbKey::ArrowDown,
-                "arrowright" => druid::KbKey::ArrowRight,
-                "arrowleft" => druid::KbKey::ArrowLeft,
-                "up" => druid::KbKey::ArrowUp,
-                "down" => druid::KbKey::ArrowDown,
-                "right" => druid::KbKey::ArrowRight,
-                "left" => druid::KbKey::ArrowLeft,
-                "tab" => druid::KbKey::Tab,
-                "enter" => druid::KbKey::Enter,
-                "delete" => druid::KbKey::Delete,
-                "del" => druid::KbKey::Delete,
-                _ => druid::KbKey::Character(parts[parts.len() - 1].to_string()),
+
+            let last_part = parts[parts.len() - 1];
+            let key =
+                KeyPressData::map_str_to_key(last_part.to_lowercase().as_str());
+            let key = if let Some(key) = key {
+                key
+            } else {
+                // Skip past unrecognized key definitions
+                eprintln!("Unrecognized key: {}", last_part);

```

Comment:
👍 
Nice for debugging.
Do you think its worth ticketing up introducing a logging library, or is standard prints probably enough for an IDE?

Reply:
Optimally, this would be a notification in the corner like VSCode has. That way we'd inform the user that there was an issue, but I don't think Lapce has an existing way to make those so I didn't do that here.    
`tracing` could be added without dependency cost, since we already depend on druid which uses tracing itself. I think it could be useful, especially with settings/environment variables to tell it to log certain things (so then we don't have to wait out the long rebuild times after we add a few prints).   
Overall, I think it could provide some benefit but isn't too major.

Reply:
You can use ```log::warn!() ```

And change the log level here https://github.com/lapce/lapce/blob/master/lapce-ui/src/app.rs#L19

which will be logged to a file in the config folder.